### PR TITLE
Add templates and third_party Bootstrap and Font Awesome assets.

### PR DESCRIPTION
**Pull request by davidzchen**
* Add Jekyll template files
* Add third_party Bootstrap and Font Awesome files
* Add serve-docs.sh script
* Add BUILD and WORKSPACE files and set up site build

### CODE REVIEW

File: _includes/head.html [hunk 1]
```diff
@@ -0,0 +1,62 @@
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <title>{{ page.title }} - Bazel</title>
+
+    <script>
+      var current_url = window.location.href;
```
Comment (steren) [inline _includes/head.html]: since the blog will live at blog.bazel.build, we do not need this redirect code

File: _includes/footer.html [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    <script>
+      var shiftWindow = function() {
+        if (location.hash.length !== 0) {
+          window.scrollBy(0, -50);
+        }
+      };
+      window.addEventListener("hashchange", shiftWindow);
+
+      var highlightCurrentSidebarNav = function() {
```
Comment (steren) [inline _includes/footer.html]: Not sure we needthis logic, as the blog does not have a nav

File: _includes/head.html [hunk 3]
```diff
@@ -0,0 +1,62 @@
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <title>{{ page.title }} - Bazel</title>
```
Comment (steren) [inline _includes/head.html]: use "- Bazel blog"

File: _layouts/default.html [hunk 4]
```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
```
Comment (steren) [inline _layouts/default.html]: Do we need this default layout?

Comment (davidzchen) [inline _layouts/default.html]: The 404 page uses it.

File: _includes/head.html (hunk 3 above)
Comment (davidzchen) [inline _includes/head.html]: Done.

File: _includes/head.html (hunk 1 above)
Comment (davidzchen) [inline _includes/head.html]: Removed all the redirect code except for the http://blog.bazel.build -> https://blog.bazel.build redirect.

File: _includes/footer.html (hunk 2 above)
Comment (davidzchen) [inline _includes/footer.html]: Removed.

**Pull request merged**